zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/metrics/CamundaExporterMetrics.java

```diff
@@ -164,6 +181,19 @@ public void recordBatchOperationsArchiving(final int count) {
     batchOperationsArchiving.increment(count);
   }
 
+  /**
+   * For each record write timestamp, observes the export latency by subtracting the timestamp from
+   * the current stream clock.
+   *
+   * @param recordTimestamps a collection of {@link Record#getTimestamp()}
+   */
+  public void observeRecordExportLatencies(final Collection<Long> recordTimestamps) {
```

Comment:
:thought_balloon: I pushed the logic of computing the latency from record timestamps using the stream clock into the metrics. We could also do that in the writer and keep the metrics class dumb, but I didn't like pushing too much state into the writer just for that.